contracts/contracts/strategies/balancer/BalancerMetaPoolStrategy.sol

```diff
@@ -0,0 +1,477 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Balancer MetaStablePool Strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { BaseAuraStrategy, BaseBalancerStrategy } from "./BaseAuraStrategy.sol";
+import { IBalancerVault } from "../../interfaces/balancer/IBalancerVault.sol";
+import { IRateProvider } from "../../interfaces/balancer/IRateProvider.sol";
+import { IMetaStablePool } from "../../interfaces/balancer/IMetaStablePool.sol";
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+contract BalancerMetaPoolStrategy is BaseAuraStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        BaseBalancerConfig memory _balancerConfig,
+        address _auraRewardPoolAddress
+    )
+        InitializableAbstractStrategy(_stratConfig)
+        BaseBalancerStrategy(_balancerConfig)
+        BaseAuraStrategy(_auraRewardPoolAddress)
+    {}
+
+    /**
+     * @notice Deposits an `_amount` of vault collateral assets
+     * from the this strategy contract to the Balancer pool.
+     * @param _asset Address of the Vault collateral asset
+     * @param _amount The amount of Vault collateral assets to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        whenNotInVaultContext
```

Comment:
I don't think we need these read only reentrancy checks on deposits and withdraws, since balancer's primary reentrancy protection should be active since they are writes that interact with the balancer contracts.

Argue with me if I'm wrong. :)

Reply:
correct, I don't know what I was thinking :)